### PR TITLE
Converting json junit test to mocha.

### PR DESCRIPTION
**Pull request by moniika**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

#2870

### Proposed Changes
Convert json jsunit test to mocha.

### Reason for Changes
Explained in linked issue.

### Test Coverage
Ran mocha tests.

### CODE REVIEW

File: tests/mocha/json_test.js [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+      assertEquals(Blockly.FieldDropdown, dropdown.constructor);
+      assertEquals(VALUE0, dropdown.getValue());
+
+      var options = dropdown.getOptions();
+      var image0 = options[0][0];
+      assertEquals(IMAGE0.width, image0.width);
+      assertEquals(IMAGE0.height, image0.height);
+      assertEquals(IMAGE0.src, image0.src);
+      assertEquals(IMAGE0.alt, image0.alt);
+      assertEquals(VALUE0, options[0][1]);
```
Comment (alschmiedt) [inline tests/mocha/json_test.js:326]: Could this be pulled out into a helper assert functino?

File: tests/mocha/json_test.js [hunk 2]
```diff
@@ -0,0 +1,343 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
```
Comment (alschmiedt) [inline tests/mocha/json_test.js:3]: Update to 2020

File: tests/mocha/json_test.js [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+      var block = this.blocks_[i];
+      block.dispose();
+    }
+    for (var i = 0, blockType; (blockType = this.blockTypes_[i]); i++) {
+      delete Blockly.Blocks[blockType];
+    }
+    for (var i = 0, message; (message = this.messages_[i]); i++) {
+      delete Blockly.Msg[message];
+    }
+    this.workspace_.dispose();
```
Comment (alschmiedt) [inline tests/mocha/json_test.js:26]: Not really for this PR, but might be good to see if this is needed in other tests. If so, would be good to pull out into a function for use throughout all tests.

File: tests/mocha/json_test.js [hunk 4]
```diff
@@ -0,0 +1,343 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+suite('JSON', function() {
```
Comment (alschmiedt) [inline tests/mocha/json_test.js:7]: I know this is probably copied over from jsunit tests, but I think a more descriptive name would be better here. Possibly JSON Block Definitions?

File: tests/mocha/json_test.js (hunk 3 above)
Comment (moniika) [reply tests/mocha/json_test.js:26]: I'm working out what exactly this should look like but I'm fairly certain I'm going to add this functionality as a helper function that all the tests can call.

Comment (moniika) [reply tests/mocha/json_test.js:26]: For future PR.

File: tests/mocha/json_test.js (hunk 4 above)
Comment (moniika) [reply tests/mocha/json_test.js:7]: Done.

File: tests/mocha/json_test.js (hunk 1 above)
Comment (moniika) [reply tests/mocha/json_test.js:326]: Done.

File: tests/mocha/json_test.js (hunk 2 above)
Comment (moniika) [reply tests/mocha/json_test.js:3]: Done.

**Pull request merged**